src/main/java/frc/team2412/robot/commands/autonomous/SwerveControllerCommand.java

```diff
@@ -0,0 +1,189 @@
+    private final Supplier<Pose2d> pose;
+    private final SwerveDriveKinematics kinematics;
+    private final HolonomicDriveController controller;
+    private final Consumer<ChassisSpeeds> outputModuleStates;
+    private final Supplier<Rotation2d> desiredRotation;
+    private final DrivebaseSubsystem drivebase;
+
+    /**
+     * Constructs a new SwerveControllerCommand that when executed will follow the
+     * provided
+     * trajectory. This command will not return output voltages but rather raw
+     * module states from the
+     * position controllers which need to be put into a velocity PID.
+     *
+     * <p>
+     * Note: The controllers will *not* set the outputVolts to zero upon completion
+     * of the path-
+     * this is left to the user, since it is not appropriate for paths with
+     * nonstationary endstates.
+     *
+     * @param trajectory
+     *            The trajectory to follow.
+     * @param pose
+     *            A function that supplies the robot pose - use one of the odometry
+     *            classes to
+     *            provide this.
+     * @param kinematics
+     *            The kinematics for the robot drivetrain.
+     * @param xController
+     *            The Trajectory Tracker PID controller for the robot's x position.
+     * @param yController
+     *            The Trajectory Tracker PID controller for the robot's y position.
+     * @param thetaController
+     *            The Trajectory Tracker PID controller for angle for the robot.
+     * @param desiredRotation
+     *            The angle that the drivetrain should be facing. This is sampled at
+     *            each
+     *            time step.
+     * @param outputModuleStates
+     *            The raw output module states from the position controllers.
+     * @param drivebase
+     * @param requirements
+     *            The subsystems to require.
+     */
+    @SuppressWarnings("ParameterName")
+    public SwerveControllerCommand(
+            Trajectory trajectory,
+            Supplier<Pose2d> pose,
+            SwerveDriveKinematics kinematics,
+            PIDController xController,
+            PIDController yController,
+            ProfiledPIDController thetaController,
+            Supplier<Rotation2d> desiredRotation,
+            Consumer<ChassisSpeeds> outputModuleStates,
+            DrivebaseSubsystem drivebase, Subsystem... requirements) {
+        this.trajectory = requireNonNullParam(trajectory, "trajectory", "SwerveControllerCommand");
+        this.pose = requireNonNullParam(pose, "pose", "SwerveControllerCommand");
+        this.kinematics = requireNonNullParam(kinematics, "kinematics", "SwerveControllerCommand");
+        this.drivebase = drivebase;
+
+        controller = new HolonomicDriveController(
+                requireNonNullParam(xController, "xController", "SwerveControllerCommand"),
+                requireNonNullParam(yController, "xController", "SwerveControllerCommand"),
```

Comment:
copy/paste error - should be yController both places

Reply:
good catch